sqldatabaseproductserver/main.go

```diff
@@ -0,0 +1,196 @@
+package main
+
+import (
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"github.com/gorilla/mux"
+	_ "github.com/mattn/go-sqlite3"
+	"io"
+	"log"
+	"net/http"
+	"strconv"
+)
+
+type Product struct {
+	ID    int    `json:"id"`
+	Name  string `json:"name"`
+	Price int    `json:"price"`
+}
+
+var db *sql.DB
+var err error
+
+func main() {
+	db, err = sql.Open("sqlite3", "products.db")
+	if err != nil {
+		fmt.Println("error opening database")
+		return
+	}
+	err = db.Ping()
+	if err != nil {
+		fmt.Println("error pinging database")
+		return
+	}
+	r := mux.NewRouter()
+	r.HandleFunc("/products", GetProducts).Methods("GET")
+	r.HandleFunc("/products/{id}", GetProduct).Methods("GET")
+	r.HandleFunc("/products", AddProduct).Methods("POST")
+	r.HandleFunc("/products/{id}", UpdateProduct).Methods("PUT")
+	r.HandleFunc("/products/{id}", DeleteProduct).Methods("DELETE")
+	fmt.Println("Server is running")
+	log.Fatal(http.ListenAndServe("localhost:8080", r))
+
+}
+
+func GetProducts(w http.ResponseWriter, r *http.Request) {
+	var products []Product
+	limit := r.FormValue("limit")
+	query := "SELECT * FROM products LIMIT " + limit + ";"
```

Comment:
You're opening yourself up to SQL injection attacks with this. Use prepared statements for your queries.